operators/pkg/controller/elasticsearch/reconcile/state.go

```diff
@@ -157,3 +157,13 @@ func (s *State) UpdateElasticsearchInvalid(results []validation.Result) {
 		s.AddEvent(corev1.EventTypeWarning, events.EventReasonValidation, r.Reason)
 	}
 }
+
+// UpdateElasticsearchControllerVersion sets the elasticsearch controller version that last updated the ES cluster
+func (s *State) UpdateElasticsearchControllerVersion(version string) {
+	s.status.ControllerVersion = version
+}
+
+// GetElasticsearchControllerVersion returns the elasticsearch operator version that last updated the ES cluster
+func (s *State) GetElasticsearchControllerVersion() string {
```

Comment:
not sure about `controller` vs. `operator` 🤷‍♂ 
I don't have strong opinion. Maybe some else?

Reply:
In my mind the operator is a set of controllers so I find `controller` fine.
https://stackoverflow.com/a/47857073/10614521